pkg/schema/util_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package schema_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/aws/aws-service-operator-k8s/pkg/names"
+	"github.com/aws/aws-service-operator-k8s/pkg/testutil"
+)
+
+func TestGoTypeFromSchemaRef(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	tests := []struct {
```

Comment:
Are these all happy paths / will there be tests intentionally leading to the placeholder types or return values (the "unknown" strings) added later? Or do those parts never get executed?

Reply:
they are indeed all the happy paths, good catch, Alina! I can add some of the unhappy paths to the tests.